packages/core/src/controllers/relayer.ts

```diff
@@ -131,11 +139,13 @@ export class Relayer extends IRelayer {
   }
 
   get connected() {
-    return this.provider.connection.connected;
+    // @ts-ignore
+    return this.provider?.connection?.socket?.readyState === 1;
```

Comment:
can we use `expect-error` and explain why?

Reply:
If the TS issue is over mixed return types, can we use logic that will always return a boolean? E.g. `?? false`